### PR TITLE
[Finishes #187355116] a buyer should be able to review products

### PR DESCRIPTION
**Pull request by jniyonkuru**
**1. What does this PR do?**
This PR  enable users{buyers} to review and comment on the product they have bought

**2. Why is this important?**

this will help users who wants to buy products, to get other buyers experience  with the product , and will also help the seller to understand how their product are perceived

**3. How should this be manually tested?**
Clone the repository.
Checkout to the branch.
Install necessary dependencies.
Start the development server.
Utilize Swagger documentation to interact with the API.
**Screenshots**
Screenshot from 2024-06-04 10-53-17
Screenshot from 2024-06-04 10-52-41

### CODE REVIEW

**Review by lilika67 — APPROVED** [review]

File: src/controllers/productController.ts [hunk 1]
```diff
@@ -19,6 +19,7 @@ export interface CustomRequest extends Request {
 }
 
 export async function createCollection(req: CustomRequest, res: Response) {
+  console.log(req.body);
```
Comment (UmuhireJessie) [inline src/controllers/productController.ts]: Remove this console

File: src/controllers/ProductReviewController.ts [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+    validateProductReview(req.body as Review);
+    const user = (req as any).user;
+    try {
+      const foundUser = await db.User.findOne({
+        where: { userId: user.userId },
+      });
+
+      if (!foundUser) {
+        return res.status(404).json({
+          message: 'user with the given is not found',
```
Comment (UmuhireJessie) [inline src/controllers/ProductReviewController.ts]: What do you mean by this message?

**Review by UmuhireJessie — CHANGES_REQUESTED** [review]
Check my comments and resolve the conflicts

File: src/controllers/ProductReviewController.ts (hunk 2 above)
Comment (jniyonkuru) [reply src/controllers/ProductReviewController.ts]: the id word is missing

**Review by shema-surge — APPROVED** [review]

File: src/controllers/ProductReviewController.ts (hunk 2 above)
Comment (UmuhireJessie) [reply src/controllers/ProductReviewController.ts]: Can you make it clear in the message because we can not know what you meant by "the given"

**Review by UmuhireJessie — APPROVED** [review]

**Pull request merged**